test/test_shared.cpp

```diff
@@ -3815,4 +3815,31 @@ TEST(Shared_LockFileOfWrongCondvarSizeThrows)
 }
 
 
+// Test if we can successfully open an existing encrypted file (generated by Core 4.0.3)
+TEST(Shared_DecryptExisting)
```

Comment:
I think you need `TEST_IF(Shared_DecryptExisting, REALM_ENABLE_ENCRYPTION)` otherwise we'll fail when running tests without encryption compiled in.